frappe/integrations/doctype/ldap_settings/ldap_settings.py

```diff
@@ -370,10 +370,12 @@ def login():
 	args = frappe.form_dict
 	ldap: LDAPSettings = frappe.get_doc("LDAP Settings")
 
-	user = ldap.authenticate(frappe.as_unicode(args.usr), frappe.as_unicode(args.pop("pwd", None)))
+	password = args.pop("pwd", None)
```

Comment:
Password is deliberately popped from the form dict here. Adding it back reverses this: 

https://github.com/frappe/frappe/pull/17872

Reply:
Yes but appearantly now ldap users cant create a 2fa token anymore. Perhaps then just the dummy string insertion ?

Reply:
can you share error/traceback that happens when ldap and 2fa are enabled together?

Reply:
Adding dummy string is weird solution, we should handle it better if possible.

Reply:
When NO token exists then authenticate_for_2factor(user.name) calls a temp token :

/home/frappe/frappe-bench/apps/frappe/frappe/twofactor.py: 92
90 	token = int(pyotp.TOTP(otp_secret).now())
91 	tmp_id = frappe.generate_hash(length=8)
92 	cache_2fa_data(user, token, otp_secret, tmp_id)
93 	verification_obj = get_verification_obj(user, token, otp_secret)
94 	# Save data in local

cache_2fa_data is called which makes a temp token : 
110 		expiry_time = frappe.flags.otp_expiry or 180
111 	for k, v in iteritems({"_usr": user, "_pwd": pwd, "_otp_secret": otp_secret}):
112 		frappe.cache().set("{0}{1}".format(tmp_id, k), v)
113 		frappe.cache().expire("{0}{1}".format(tmp_id, k), expiry_time) 

the problem is when iteritems takes _pwd -> this is then a NoneType on which the error occurs.
As far as i can see when checking the token this is no longer rechecked for LDAP (it is for normal logon) 
so a dummy string would be possible.